src/basic/REDUCE3_INT-OMP.cpp

```diff
@@ -91,22 +91,53 @@ void REDUCE3_INT::runOpenMPVariant(VariantID vid, size_t RAJAPERF_UNUSED_ARG(tun
 
     case RAJA_OpenMP : {
 
-      startTimer();
-      for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
+      if (tune_idx == 0) {
+
+        startTimer();
+        for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
+
+          RAJA::ReduceSum<RAJA::omp_reduce, Int_type> vsum(m_vsum_init);
+          RAJA::ReduceMin<RAJA::omp_reduce, Int_type> vmin(m_vmin_init);
+          RAJA::ReduceMax<RAJA::omp_reduce, Int_type> vmax(m_vmax_init);
+  
+          RAJA::forall<RAJA::omp_parallel_for_exec>(
+            RAJA::RangeSegment(ibegin, iend), [=](Index_type i) {
+            REDUCE3_INT_BODY_RAJA;
+          });
+
+          m_vsum += static_cast<Int_type>(vsum.get());
+          m_vmin = RAJA_MIN(m_vmin, static_cast<Int_type>(vmin.get()));
+          m_vmax = RAJA_MAX(m_vmax, static_cast<Int_type>(vmax.get()));
+
+        }
+        stopTimer();
+
+      } 
 
-        RAJA::ReduceSum<RAJA::omp_reduce, Int_type> vsum(m_vsum_init);
-        RAJA::ReduceMin<RAJA::omp_reduce, Int_type> vmin(m_vmin_init);
-        RAJA::ReduceMax<RAJA::omp_reduce, Int_type> vmax(m_vmax_init);
+      if (tune_idx == 1) {
 
-        RAJA::forall<RAJA::omp_parallel_for_exec>(
-          RAJA::RangeSegment(ibegin, iend), [=](Index_type i) {
-          REDUCE3_INT_BODY_RAJA;
-        });
+        startTimer();
+        for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
 
-        m_vsum += static_cast<Int_type>(vsum.get());
-        m_vmin = RAJA_MIN(m_vmin, static_cast<Int_type>(vmin.get()));
-        m_vmax = RAJA_MAX(m_vmax, static_cast<Int_type>(vmax.get()));
+          Int_type tvsum = m_vsum_init;
+          Int_type tvmin = m_vmin_init;
+          Int_type tvmax = m_vmax_init;
 
+          RAJA::forall<RAJA::omp_parallel_for_exec>(
+            RAJA::RangeSegment(ibegin, iend),
+            RAJA::expt::Reduce<RAJA::operators::plus>(&tvsum),
+            RAJA::expt::Reduce<RAJA::operators::minimum>(&tvmin),
+            RAJA::expt::Reduce<RAJA::operators::maximum>(&tvmax),
+            [=](Index_type i, Int_type& vsum, Int_type& vmin, Int_type& vmax) {
+              REDUCE3_INT_BODY;
+            }
+          );
+
+          m_vsum += static_cast<Int_type>(tvsum);
```

Comment:
Why cast here?

Reply:
It's probably not needed. I was trying to be explicit and follow the pattern Mike established in the RAJA tests for the new reduction stuff.